components/coreapi/blocks.go

```diff
@@ -28,26 +33,100 @@ func blockMetadataResponseByID(c echo.Context) (*blockMetadataResponse, error) {
+		BlockID:            block.ID().ToHex(),
+		StrongParents:      block.Block().StrongParents.ToHex(),
+		WeakParents:        block.Block().WeakParents.ToHex(),
+		ShallowLikeParents: block.Block().ShallowLikeParents.ToHex(),
+		BlockState:         blockStatePending.String(),
 	}
-	_, exists := deps.Protocol.MainEngineInstance().Block(block.ID())
-	if !exists {
-		bmResponse.BlockError = "block not found"
-	}
-
-	// todo set states and error
 
 	return bmResponse, nil
 }
 
 func blockIssuance(_ echo.Context) (*blockIssuanceResponse, error) {
-	//nolint:nilnil // temporary nil,nil
-	return nil, nil
+	references := deps.Protocol.TipManager.Tips(iotago.BlockMaxParents)
+	slotCommitment := deps.Protocol.MainEngineInstance().Storage.Settings().LatestCommitment()
+
+	if len(references[model.StrongParentType]) == 0 {
+		return nil, errors.Errorf("could not get references")
+	}
+
+	cBytes, err := deps.Protocol.API().JSONEncode(slotCommitment.Commitment())
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &blockIssuanceResponse{
+		StrongParents:       references[model.StrongParentType].ToHex(),
+		WeakParents:         references[model.WeakParentType].ToHex(),
+		ShallowLikeParents:  references[model.ShallowLikeParentType].ToHex(),
+		LatestFinalizedSlot: deps.Protocol.MainEngineInstance().Storage.Settings().LatestFinalizedSlot(),
+		Commitment:          cBytes,
+	}
+
+	return resp, nil
 }
 
-func sendBlock(_ echo.Context) (*blockCreatedResponse, error) {
-	//nolint:nilnil // temporary nil,nil
-	return nil, nil
+func sendBlock(c echo.Context) (*blockCreatedResponse, error) {
+	mimeType, err := httpserver.GetRequestContentType(c, httpserver.MIMEApplicationVendorIOTASerializerV1, echo.MIMEApplicationJSON)
+	if err != nil {
+		return nil, err
+	}
+
+	iotaBlock := &iotago.Block{
+		Signature: &iotago.Ed25519Signature{},
+	}
+
+	switch mimeType {
+	case echo.MIMEApplicationJSON:
+		if err := c.Bind(iotaBlock); err != nil {
+			return nil, errors.WithMessagef(httpserver.ErrInvalidParameter, "invalid block, error: %s", err)
+		}
+
+	case httpserver.MIMEApplicationVendorIOTASerializerV1:
+		if c.Request().Body == nil {
+			// bad request
+			return nil, errors.WithMessage(httpserver.ErrInvalidParameter, "invalid block, error: request body missing")
+		}
+
+		bytes, err := io.ReadAll(c.Request().Body)
+		if err != nil {
+			return nil, errors.WithMessagef(httpserver.ErrInvalidParameter, "invalid block, error: %s", err)
+		}
+
+		// Do not validate here, the parents might need to be set
+		if _, err := deps.Protocol.API().Decode(bytes, iotaBlock); err != nil {
+			return nil, errors.WithMessagef(httpserver.ErrInvalidParameter, "invalid block, error: %s", err)
+		}
+
+	default:
+		return nil, echo.ErrUnsupportedMediaType
+	}
+
+	mergedCtx, mergedCtxCancel := contextutils.MergeContexts(c.Request().Context(), Component.Daemon().ContextStopped())
+	defer mergedCtxCancel()
+
+	blockID, err := deps.BlockIssuer.AttachBlock(mergedCtx, iotaBlock)
```

Comment:
Here we need to handle if the node allows clients to send unfinished blocks or not. Same as with the PoW this is an opt-in feature of the node operator, since it will consume the node operators Mana if the node issues the block instead of the user.